src/mpl/include/mpl_base.h

```diff
@@ -42,7 +42,11 @@
 #endif /* ATTRIBUTE */
 
 #define MPL_UNUSED ATTRIBUTE((unused))
+#ifdef MPL_ENABLE_FORCE_INLINE
 #define MPL_STATIC_INLINE_PREFIX ATTRIBUTE((always_inline)) static inline
+#else
+#define MPL_STATIC_INLINE_PREFIX static inline
+#endif
```

Comment:
Do we have a use case that we *don't want* always_inline?

Reply:
I have an opposite opinion: if the compiler doesn't want to inline, usually we should not inline.

Reply:
> I have an opposite opinion: if the compiler doesn't want to inline, usually we should not inline.

I think I can agree to that.

Reply:
There are two issues in current main branch regarding static inline:
- Some functions use `static inline` directly, but some others use the wrapper macro `MPL_STATIC_INLINE_PREFIX `. They should be consistent as we don't distinguish functions that must be always_inline v.s. inline based on compiler's decision. 
- `MPL_STATIC_INLINE_PREFIX` always set `always_inline`. Thus for functions that use it, symbols cannot be generated for debugging even with `-O0 -g` (e.g., no debugging symbol for `MPIDI_put_unsafe`)

To fix these issues, I did two things in this PR:
- Replace direct `static inline` code with `MPL_STATIC_INLINE_PREFIX` in MPICH/ch4
- Add an option "inline" in --enable-fast to control `always_inline` in `MPL_STATIC_INLINE_PREFIX`. If "inline" is not given, the wrapper will be equal to `static inline`. By default, --enable-fast=O2,inline.

Do you think it makes sense?

Reply:
To address @shintaro-iwasaki 's opinion, I agree with it too. So maybe we should keep the default value of --enable-fast as "O2" (i.e., "inline" is disabled by default)